src/js-form-validator.ts

```diff
@@ -248,7 +256,7 @@ export class FormValidator {
       0: Not empty, but invalid.
       -1: empty
   */
-  private static validate_rule(type, value): number {
+  private static validate_rule(type, value, allowEmpty): number {
```

Comment:
Could we update all of these methods to be proper convention to our coding standards. So this would become `private static ValidateRule(type: string, value: any, allowEmpty: bool): number {`

Better even, update the `type` parameter to an enumeration.